contrib/ruff/template.py

```diff
@@ -0,0 +1,49 @@
+pkgname = "ruff"
+pkgver = "0.3.3"
+pkgrel = 0
+build_style = "python_pep517"
+hostmakedepends = [
+    "cargo",
+    "python-build",
+    "python-installer",
+    "python-maturin",
+]
+makedepends = ["rust-std"]
+pkgdesc = "Python formatter and linter"
+maintainer = "psykose <alice@ayaya.dev>"
+license = "MIT"
+url = "https://docs.astral.sh/ruff"
+source = f"https://github.com/astral-sh/ruff/archive/refs/tags/v{pkgver}.tar.gz"
+sha256 = "0065193961b1398dfdb43cc196a67f7275cc97a711b994430b62905144116e19"
+# generates completions with host bin
+options = ["!cross"]
+
+
+def post_patch(self):
+    from cbuild.util import cargo
+
+    cargo.Cargo(self).vendor()
+    cargo.setup_vendor(self)
+
+
+def post_build(self):
+    for shell in ["bash", "fish", "zsh"]:
+        with open(self.cwd / f"ruff.{shell}", "w") as f:
+            self.do(
+                "./target/release/ruff",
+                "--generate-shell-completion",
+                shell,
+                stdout=f,
+            )
+
+
+def do_check(self):
```

Comment:
is there any actual need for this though

Reply:
you mean the tests at all, or the manual check? pytest doesn't run them